src/main/java/study/Calculator.java

```diff
@@ -0,0 +1,60 @@
+package study;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Calculator {
+	public int run(String input) {
+		String[] numbers = split(input);
+		int sum = sum(numbers);
+		return sum;
+	}
+
+	private int sum(String[] numbers) {
+		int sum = 0;
+		for (int i = 0; i < numbers.length; i++) {
+			sum += validatePositiveNumber(numbers[i]);
+		}
+		return sum;
+	}
+
+	public String[] split(String input) {
+		if (isNullOrEmpty(input)) {
+			return new String[] {"0"};
+		}
+
+		return splitDelimiter(input);
+	}
+
+	public int validatePositiveNumber(String input) {
+		if (input.contains("-")) {
+			throw new RuntimeException();
+		}
+
+		try {
+			return Integer.valueOf(input);
+		} catch (NumberFormatException ex) {
+			throw new RuntimeException();
+		}
+
+	}
+
+	private String[] splitDelimiter(String input) {
+		Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(input);
```

Comment:
해당 메서드가 호출될때마다 패턴을 컴파일하도록 되어있네요!
패턴 컴파일은 시간이 많이 걸리는 프로세스이며 자원을 많이 사용하는것으로 알고 있습니다.
그렇기에 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다.
```
If a pattern is to be used multiple times, compiling it once and reusing 
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 캐싱하여 개선하면 좋을 것 같아요 :)